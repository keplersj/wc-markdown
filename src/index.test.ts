import fetchMock from "jest-fetch-mock";
import { jest } from "@jest/globals";
import { html } from "atomico";
import { fixture } from "atomico/test-dom";
import { findByText } from "@testing-library/dom";
import "./index.js";

jest.setTimeout(10000);

describe("Remark-Markdown Web Component", () => {
  it("renders basic component", async () => {
    const component = fixture(html`<remark-markdown></remark-markdown>`);

    await (component as any).updated;

    expect(component).toMatchSnapshot();
    expect(component.shadowRoot?.innerHTML).toMatchSnapshot();
  });

  beforeEach(() => {
    fetchMock.mockResponses("# Test");
  });

  it("renders component with `src` parameter", async () => {
    const component = fixture(
      html`<remark-markdown
        src="https://raw.github.com/adamschwartz/github-markdown-kitchen-sink/master/README.md"
      ></remark-markdown>`
    );

    await findByText(component.parentElement!, "Test").catch((e) =>
      console.error(e)
    );

    expect(component).toMatchSnapshot();
    expect(component.shadowRoot?.innerHTML).toMatchSnapshot();
  });
});
