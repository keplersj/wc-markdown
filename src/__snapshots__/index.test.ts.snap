// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Remark-Markdown Web Component renders basic component 1`] = `<remark-markdown />`;

exports[`Remark-Markdown Web Component renders basic component 2`] = `
<div>
</div>
`;

exports[`Remark-Markdown Web Component renders component with \`src\` parameter 1`] = `
<remark-markdown
  src="https://raw.github.com/adamschwartz/github-markdown-kitchen-sink/master/README.md"
/>
`;

exports[`Remark-Markdown Web Component renders component with \`src\` parameter 2`] = `
<div>
  <h1>
    Test
  </h1>
</div>
`;
